crates/rattler_lock/src/builder.rs

```diff
@@ -362,3 +396,77 @@ impl From<PypiPackageEnvironmentData> for HashablePypiPackageEnvironmentData {
         }
     }
 }
+
+#[cfg(test)]
+mod test {
+    use std::{collections::BTreeSet, str::FromStr};
+
+    use rattler_conda_types::{PackageName, PackageRecord, Platform, Version};
+    use url::Url;
+
+    use crate::{CondaBinaryData, LockFile};
+
+    #[test]
+    fn test_merge_records_and_purls() {
+        let record = PackageRecord {
```

Comment:
One small suggestion for the test:

maybe we could test here that indeed a package with purl will win in case of merging with one that doesn't have purl.
This is my idea: